dhis2-sre/im-manager
# fix: update rabbitmq lib to fix connection issue DEVOPS-366

* Configure it to set the connection name and consumer prefix to the pod name so we can tell it apart.
* Configure the rabbitmq lib to log using slog. See description down below
* also update Go to 1.22.2 as the toxiproxy v2 needs at least 1.22.1

## Manual test

* log in the RabbitMQ UI in the deployed app and disconnect the im-manager connection
  * you can see the connection name is the pod name
  * the consumer tag is prefixed with the pod name
* follow the logs of the im-manager to see that when you disconnect the RabbitMQ library logs using slog and the described attributes

## Slog

We are creating a logger in main. This will later https://dhis2.atlassian.net/browse/DEVOPS-170 be used to log in the im-manager itself.

We can add attributes to this "root" logger which will then be logged in every log line. Loki does not index the log entry. Loki already provides labels for pod name, namespace, app so we do not need to add such attributes. We use a text handler instead of a JSON one as Loki does not index the log line and its easier for us to read.

The RabbitMQ consumer gets a logger from the "root" one wit [group](https://pkg.go.dev/log/slog#hdr-Groups) `rabbitmq` so we do not get in naming conflicts when logging in the library. The consumers log entries will have all the entries defined by the "root" logger. The standard ones like `msg`, `time` will stay top level only the ones it adds itself like `consumerTag` will be nested under the group.

Lines from the RabbitMQ consumer will have

* connectionName=`<if configured using WithConnectionName>`

## Next

https://dhis2.atlassian.net/browse/DEVOPS-170 use slog in the rest of the manager.

## Some day

We could add an attribute with the app version. When https://github.com/golang/go/issues/51279 is fixed we can rely on the https://pkg.go.dev/runtime/debug#BuildInfo to set the version. Otherwise we could use an `ldflag` and set a global in main.
